modules/standard/Map.chpl

```diff
@@ -210,7 +210,7 @@ module Map {
 
       :returns: Reference to the value mapped to the given key.
     */
-    proc this(k: keyType) ref {
+    proc this(k: keyType) ref where !isNonNilableClass(valType) {
```

Comment:
Do we want to adjust this where clause to be `where isDefaultInitializable`? Or do we want to handle that effort later?

Up to you, I'm not entirely sure.